assembly.py

```diff
@@ -471,6 +472,52 @@ def order_and_orient(inFasta, inReference, outFasta,
     #    os.unlink(trimmed)
     return 0
 
+def _call_order_and_orient_orig(kwargs):
+    _order_and_orient_orig(**kwargs)
+
+def order_and_orient(inFasta, inReference, outFasta,
+        outAlternateContigs=None, outReference=None,
+        breaklen=None, # aligner='nucmer', circular=False, trimmed_contigs=None,
+        maxgap=200, minmatch=10, mincluster=None,
+        min_pct_id=0.6, min_contig_len=200, min_pct_contig_aligned=0.3, n_genome_segments=0, threads=0):
+    ''' This step cleans up the de novo assembly with a known reference genome.
+        Uses MUMmer (nucmer or promer) to create a reference-based consensus
+        sequence of aligned contigs (with runs of N's in between the de novo
+        contigs).
+    '''
+
+    ref_segments_all = tuple(Bio.SeqIO.parse(inReference, 'fasta'))
+    n_genome_segments = n_genome_segments or len(ref_segments_all)
+    assert (len(ref_segments_all) % n_genome_segments) == 0
+    n_refs = int(len(ref_segments_all) / n_genome_segments)
+    log.info('n_genome_segments={} n_refs={}'.format(n_genome_segments, n_refs))
+
+    with util.file.tempfnames([ '.{}.ref.fasta'.format(ref_num) for ref_num in range(n_refs)]) as refs_fasta, \
```

Comment:
Are these temp filenames hidden files (start with `.` )?

Reply:
Nevermind - it's actually a suffix!

Reply:
They're not hidden as long as the prefix is set (defaults to `tmp`, so `tmp.{}.ref.fasta`...)
https://github.com/broadinstitute/viral-ngs/blob/master/util/file.py#L116